src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -154,15 +178,16 @@ private void visitObject(MaskingState maskingState, @Nullable KeyMaskingConfig p
                 break;
             }
             // In case target keys should be considered as allow list, we need to NOT mask certain keys
-            int openingQuoteIndex = maskingState.currentIndex();
+            maskingState.registerValueStartIndex();
 
             stepOverStringValue(maskingState);
 
             int afterClosingQuoteIndex = maskingState.currentIndex();
-            int keyLength = afterClosingQuoteIndex - openingQuoteIndex - 2; // minus the opening and closing quotes
-            maskingState.expandCurrentJsonPath(keyMatcher.traverseJsonPathSegment(maskingState.getMessage(), maskingState.getCurrentJsonPathNode(), openingQuoteIndex + 1, keyLength));
-            KeyMaskingConfig keyMaskingConfig = keyMatcher.getMaskConfigIfMatched(maskingState.getMessage(), openingQuoteIndex + 1, // plus one for the opening quote
+            int keyLength = afterClosingQuoteIndex - maskingState.getCurrentValueStartIndex() - 2; // minus the opening and closing quotes
+            maskingState.expandCurrentJsonPath(keyMatcher.traverseJsonPathSegment(maskingState.getMessage(), maskingState.getCurrentJsonPathNode(), maskingState.getCurrentValueStartIndex() + 1, keyLength));
```

Comment:
I think this is also not super clear now. Can you split this in two statements with some doc explaining what is going on?

Reply:
You are right, it does look cumbersome. I will fix it.

Reply:
Done